### PR TITLE
feat: Ruby 3 migration

### PR DESCRIPTION
**Pull request by stulacy**
Ruby 2.7 has been unsupported since March 2023 while Faculty Dev have updated their gems to work with Ruby 3, so there is no reason to stay using Ruby 2.

This PR migrates over to Ruby 3.2.4.

It will require full testing before being deployed - so far the only issue has been the deprecation of `URI::encode` but there might be other issues.

### CODE REVIEW

**Comment by stulacy** [comment]
User testing hasn't identified any further issues and the export functionality (both FACSIMILE and KPP) have not been affected by the Ruby 3 migration.

@kilicomu Can you think of anything else to check?

File: app.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Bundler.require
 
-Dir['./lib/mcm/**/*.rb'].sort.each do |file|
+Dir['./lib/mcm/**/*.rb'].each do |file|
```
Comment (kilicomu) [inline app.rb:5]: For this and other comparable lines, what was the original purpose of the `sort` call?

File: views/browse.erb [hunk 2]
```diff
@@ -22,7 +22,7 @@
     <ul>
     <% @inorganic_categories.each do | cat | %>
       <li>
-          <a href=<%= "/#{@mechanism}/inorganic?category=#{URI::encode(cat)}" %>>
+          <a href=<%= "/#{@mechanism}/inorganic?category=#{CGI::escapeURIComponent(cat)}" %>>
```
Comment (kilicomu) [inline views/browse.erb:25]: Did this change come from a Ruby 2 -> 3 migration recommendation?

Do these functions work identically?

**Review by kilicomu — COMMENTED** [review]
I've added a couple of comments but nothing major.

Have you advertised potential downtime for the site as a result of this migration?

I'd advise against doing the migration just before the weekend, if that was the plan, rather at the start of a working week!

File: views/browse.erb (hunk 2 above)
Comment (stulacy) [reply views/browse.erb:25]: It was a recommendation I found [here](https://rubyreferences.github.io/rubychanges/3.2.html) (`URI.encode` = `URI.escape`). They don't behave 100% identically as `URI.encode` was more generic and designed to escape entire URLs, while `CGI.escapeURIComponent` is more specific and exactly designed for the use-case we have here, where we want to escape a portion of text to allow it to be entered into a URL

File: app.rb (hunk 1 above)
Comment (stulacy) [reply app.rb:5]: It was required to handle imports correctly in the original repo I borrowed this line from, but doesn't affect the MCM

**Comment by stulacy** [comment]
Thanks for the comments. That's a good shout about not updating on a Friday - I'll add a banner message saying there might be a slight downtime on Monday.

**Pull request merged**